packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1327,4 +1331,69 @@ export class KubernetesClient {
   public getContextsState(): Map<string, ContextState> {
     return this.contextsState.getContextsState();
   }
+
+  async execIntoContainer(
+    podName: string,
+    containerName: string,
+    onStdOut: (data: Buffer) => void,
+    onStdErr: (data: Buffer) => void,
+    onClose: () => void,
+  ): Promise<{ onStdIn: (data: string) => void; onResize: (columns: number, rows: number) => void }> {
+    let telemetryOptions = {};
+    try {
+      const ns = this.getCurrentNamespace();
+      const connected = await this.checkConnection();
+      if (!ns || !connected) {
+        throw new Error('not connected or namespace is empty');
+      }
+
+      const stdout: ExecStreamWriter = new ResizeableTerminalWriter(new BufferedStreamWriter(onStdOut));
+      const stderr: ExecStreamWriter = new ResizeableTerminalWriter(new BufferedStreamWriter(onStdErr));
+      const stdin: StringLineReader = new StringLineReader();
+
+      const exec = new Exec(this.kubeConfig);
+      const conn = await exec.exec(
+        ns,
+        podName,
+        containerName,
+        ['/bin/sh', '-c', 'if command -v bash >/dev/null 2>&1; then bash; else sh; fi'],
+        stdout,
+        stderr,
+        stdin,
+        true,
+        (_: V1Status) => {
+          // need to think, maybe it would be better to pass exit code to the client, but on the other hand
```

Comment:
comment / still need to implement here?

Reply:
I'd rather say, it simply a comment to improve handling exit signals in future, at this moment everything works fine